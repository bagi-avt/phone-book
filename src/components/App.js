import React from "react";
import { Input, Button } from "antd";
import { PlusOutlined } from "@ant-design/icons";
import UserListContainer from "../containers/UserListContainer";
import AddUserContainer from "../containers/AddUserContainer";

const App = ({ search, inputSearchValue, checkAddUser, onClickAddUser }) => {
    const onChange = (e) => {
        inputSearchValue(e.target.value);
    };
    return (
        <div className="container">
            <h1>Phone book!</h1>
            <Input
                className="search"
                placeholder="Поиск"
                value={search}
                onChange={onChange}
            />
            <UserListContainer />
            {checkAddUser && <AddUserContainer />}
            <Button
                className="button-plus"
                type="primary"
                shape="circle"
                icon={<PlusOutlined />}
                size="large"
                onClick={onClickAddUser}
            />
        </div>
    );
};

export default App;
