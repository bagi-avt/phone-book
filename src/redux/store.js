import { createStore, combineReducers, applyMiddleware } from "redux";
import userReducer from "./user-reducer";
import thunk from "redux-thunk";

const reducers = combineReducers({
    userPage: userReducer,
});
const store = createStore(reducers, applyMiddleware(thunk));
export default store;
