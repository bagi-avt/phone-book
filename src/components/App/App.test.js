import React from "react";
import { create } from "react-test-renderer";
import App from "./App";
import store from "../../redux/store";
import {Provider} from "react-redux";

describe("User item component", () => {
    test("Matches the snapshot", () => {
        const component = create(<Provider store={store}><App/></Provider>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
