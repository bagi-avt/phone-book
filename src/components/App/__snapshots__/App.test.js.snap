// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User item component Matches the snapshot 1`] = `
<div
  className="container"
>
  <h1>
    Phone book!
  </h1>
  <input
    className="ant-input search"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="Поиск"
    type="text"
    value=""
  />
  <div />
  <button
    className="ant-btn button-plus ant-btn-primary ant-btn-circle ant-btn-lg ant-btn-icon-only"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-label="plus"
      className="anticon anticon-plus"
      role="img"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="plus"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <defs>
          <style />
        </defs>
        <path
          d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
        />
        <path
          d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
        />
      </svg>
    </span>
  </button>
</div>
`;
